graylog2-server/src/main/java/org/graylog2/migrations/MigrationHelpers.java

```diff
@@ -86,10 +86,18 @@ public String ensureBuiltinRole(String roleName, String description, Set<String>
     @Nullable
     public String ensureUser(String userName, String password, String firstName, String lastName, String email,
                              Set<String> expectedRoles) {
+        return ensureUser(userName, password, firstName, lastName, email, expectedRoles, false);
+    }
+
+    @Nullable
+    public String ensureUser(String userName, String password, String firstName, String lastName, String email,
+                             Set<String> expectedRoles, boolean isServiceAccount) {
         User previousUser = null;
         try {
             previousUser = userService.load(userName);
-            if (previousUser == null || !previousUser.getRoleIds().containsAll(expectedRoles)) {
+            if (previousUser == null
+                    || !previousUser.getRoleIds().containsAll(expectedRoles)
+                    || (isServiceAccount && !previousUser.isServiceAccount())) {
                 final String msg = "Invalid user '" + userName + "', fixing it.";
                 LOG.error(msg);
                 throw new IllegalArgumentException(msg);
```

Comment:
I realize this isn't part of your changes, but just for my own curiosity, is there a benefit from throwing an exception and immediately ignoring it in the following lines? Any reason not to remove the try-catch and just handle the logic in lines 108-122 here instead of in the catch?